configure.in

```diff
@@ -58,6 +58,9 @@ SAVE_CFLAGS="$CFLAGS"
 AC_PROG_CC([$($PG_CONFIG --cc)])
 CFLAGS="$SAVE_CFLAGS"
 
+# Add the Postgres PKGINCLUDEDIR to the include search path
+CPPFLAGS="$CPPFLAGS -I$($PG_CONFIG --pkgincludedir)"
+
```

Comment:
On 2016/03/30 11:18, Andres Freund wrote:
>> @@ -58,6 +58,9 @@ SAVE_CFLAGS="$CFLAGS"
>>  AC_PROG_CC([$($PG_CONFIG --cc)])
>>  CFLAGS="$SAVE_CFLAGS"
>>  
>> +# Add the Postgres PKGINCLUDEDIR to the include search path
>> +CPPFLAGS="$CPPFLAGS -I$($PG_CONFIG --pkgincludedir)"
>> +
> 
> Two things:
> a) for the benefit of odd platforms^WOSX we include the generated configure script in git (as does postgres).
> b) I'm a bit confused about having to play with cppflags that way. postgres' Makefile.global.in, included via pgxs.mk, included via citus'  Makefile.global.in has:
> ```
> pkgincludedir := $(shell $(PG_CONFIG) --pkgincludedir)
> ...
> includedir_server = $(pkgincludedir)/server
> includedir_internal = $(pkgincludedir)/internal
> ...
> CPPFLAGS = @CPPFLAGS@
> 
> ifdef PGXS
> override CPPFLAGS := -I$(includedir_server) -I$(includedir_internal) $(CPPFLAGS)
> else # not PGXS
> ```
> which files aren't found without the above?  I'd wager it's failures in connection_test.c? Per #411 I think the actual issue there is that internal/ has mistakenly been added to the include path.

This is what was causing the compilation to fail:

% grep -r 'internal/c.h' .
./src/backend/distributed/test/connection_utils.c:#include "internal/c.h"

This seems to work -- at least, it compiles:

{{{
% git diff
diff --git a/configure.in b/configure.in
index 09aea8f..209c292 100644
--- a/configure.in
+++ b/configure.in
@@ -58,9 +58,6 @@ SAVE_CFLAGS="$CFLAGS"
 AC_PROG_CC([$($PG_CONFIG --cc)])
 CFLAGS="$SAVE_CFLAGS"

-# Add the Postgres PKGINCLUDEDIR to the include search path
-CPPFLAGS="$CPPFLAGS -I$($PG_CONFIG --pkgincludedir)"
-
 # check for a number of CFLAGS that make development easier

 # CITUSAC_PROG_CC_CFLAGS_OPT
diff --git a/src/backend/distributed/test/connection_utils.c
b/src/backend/distributed/test/connection_utils.c
index b5048cc..e944c1f 100644
--- a/src/backend/distributed/test/connection_utils.c
+++ b/src/backend/distributed/test/connection_utils.c
@@ -13,9 +13,9 @@
  *-------------------------------------------------------------------------
  */

-#include "internal/c.h"
+#include "c.h"
 #include "libpq-fe.h"
-#include "internal/libpq-int.h"
+#include "libpq-int.h"

 #include "distributed/test_helper_functions.h"

}}}

Let me push that to my branch and update the pull request.

	Cheers,

	Matthew